browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -568,6 +568,23 @@ IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, SubFrame) {
   EXPECT_EQ(browser()->profile()->GetPrefs()->GetUint64(kAdsBlocked), 2ULL);
 }
 
+// Requests made by a service worker should be blocked as well.
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, ServiceWorkerRequest) {
+  UpdateAdBlockInstanceWithRules("adbanner.js");
+  EXPECT_EQ(browser()->profile()->GetPrefs()->GetUint64(kAdsBlocked), 0ULL);
+
+  GURL url = embedded_test_server()->GetURL(kAdBlockTestPage);
+  ui_test_utils::NavigateToURL(browser(), url);
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+
+  ASSERT_EQ(true, EvalJs(contents,
+                         "setExpectations(0, 0, 0, 1);"
+                         "installBlockingServiceWorker()"));
+  // TODO - Unclear why this doesn't increment kAdsBlocked preference
+  //EXPECT_EQ(browser()->profile()->GetPrefs()->GetUint64(kAdsBlocked), 1ULL);
```

Comment:
It appears that ads blocked from a service worker don't increment the `kAdsBlocked` preference. I don't know why, but I propose merging the main fix here for now and filing a follow-up issue for the counter.

Reply:
it is actually pretty straightforward: logic that increments counters sits in `BraveShieldsWebContentsObserver::DispatchBlockedEvent`, so it in't triggered for SW, because they have neither webcontents, nor frames.

Probably we can move the counters machinery directly to `brave_ad_block_tp_network_delegate_helper.cc` to fix this

Reply:
(let's do a f/u though)

Reply:
Awesome, that makes sense! Filed https://github.com/brave/brave-browser/issues/14087.